### PR TITLE
Acados solver code generation improvements

### PR DESCRIPTION
**Pull request by schwieni**
This is the follow-up of PR  #726
The original was not granular enough and had still some bugs.

### CODE REVIEW

**Review by FreyJo — COMMENTED** [review]
Hi @schwieni
thanks for the revised PR!
I only have 1 minor comment/question.

**Comment by schwieni** [comment]
@FreyJo regarding the Null assignment that is a good question. I believe that is a residual of some testing I did to check if the memory is still in use, etc.
I can just remove the lines in the code if you want.

File: interfaces/acados_template/acados_template/c_templates_tera/acados_solver.in.c [hunk 1]
```diff
@@ -1075,33 +1152,50 @@ int {{ model.name }}_acados_create(nlp_solver_capsule * capsule)
         ocp_nlp_cost_model_set(nlp_config, nlp_dims, nlp_in, i, "zl", zl);
         ocp_nlp_cost_model_set(nlp_config, nlp_dims, nlp_in, i, "zu", zu);
     }
+    free(zlumem);
+    Zl = NULL;
+    Zu = NULL;
+    zl = NULL;
+    zu = NULL;
```
[the remark this answers is not in the document]
Comment (schwieni) [reply interfaces/acados_template/acados_template/c_templates_tera/acados_solver.in.c]: > why are you assigning `NULL` (only) here?
> It is at least not consistent with the other pointers to heap allocated memory.

FYI: I changed that in the last commit c7a617f

Are there other points where you would like to see a change?

**Comment by FreyJo** [comment]
Thanks, merged :+1:

**Comment by schwieni** [comment]
Thank you! :-)

**Pull request merged**